src/main/capi/arrow-c.cpp

```diff
@@ -439,30 +433,16 @@ duckdb_state Ingest(duckdb_connection connection, const char *table_name, struct
 duckdb_state duckdb_arrow_scan(duckdb_connection connection, const char *table_name, duckdb_arrow_stream arrow) {
 	auto stream = reinterpret_cast<ArrowArrayStream *>(arrow);
 
-	// Backup release functions - we nullify children schema release functions because we don't want to release on
-	// behalf of the caller, downstream in our code. Note that Arrow releases target immediate children, but aren't
-	// recursive. So we only back up immediate children here and restore their functions.
 	ArrowSchema schema;
 	if (stream->get_schema(stream, &schema) == DuckDBError) {
 		return DuckDBError;
 	}
 
-	typedef void (*release_fn_t)(ArrowSchema *);
-	std::vector<release_fn_t> release_fns(duckdb::NumericCast<idx_t>(schema.n_children));
-	for (idx_t i = 0; i < duckdb::NumericCast<idx_t>(schema.n_children); i++) {
-		auto child = schema.children[i];
-		release_fns[i] = child->release;
-		child->release = arrow_array_stream_wrapper::EmptySchemaRelease;
+	if (schema.release) {
```

Comment:
We still need to ask for a schema here so we can do this check. There is a test that tries to make a null view via `duckdb_arrow_array_scan` (which passes nullptr through via the schema field of the Arrow stream). If we can release here, we should.